### PR TITLE
Fix false positive involving type parameter @Nullable annotations.

### PR DESCRIPTION
**Pull request by lazaroclapp**
NullAway doesn't currently support `@Nullable` annotations on type
parameters. However, we had an error where a `@Nullable` annotation
in a type parameter of the return type of a method was being
interpreted as applying to the method return itself.

For example:

```
public static <R> Foo<@Nullable R> bar() { ... }

public static void baz() {
  bar().toString();
}
```

would mistakenly be considered a null dereference.

We introduce an additional check to `NullabilityUtil.getTypeUseAnnotations(...)`
which will retrieve the method return annotation for the following cases:

```
@Nullable public static <R> Foo<R> bar()
@Nullable public static <R> Foo<@Nullable R> bar()
public static <R> @Nullable Foo<R> bar()
```

Yet ignore the annotation inside the type parameter of `Foo`.

It is possible we might need to refactor this check when we actually start
handling nullability of type parameters.

### CODE REVIEW

File: nullaway/src/test/java/com/uber/nullaway/NullAwayTypeUseAnnotationTests.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            "      public R get() { return null; }",
+            "    };",
+            "  }",
+            "  public static String test1() {",
+            "    // BUG: Diagnostic contains: dereferenced expression getNullableSupplierOfNullable() is @Nullable",
+            "    return getNullableSupplierOfNullable().toString();",
+            "  }",
+            "  public static String test2() {",
+            "    // The supplier contains null, but isn't itself nullable. Check against a past FP",
+            "    return getNonNullSupplierOfNullable().toString();",
```
Comment (msridhar) [inline nullaway/src/test/java/com/uber/nullaway/NullAwayTypeUseAnnotationTests.java:33]: 👍

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java [hunk 2]
```diff
@@ -271,11 +271,17 @@ public static Stream<? extends AnnotationMirror> getAllAnnotationsForParameter(
                         && t.position.parameter_index == paramInd));
   }
 
-  private static Stream<? extends AnnotationMirror> getTypeUseAnnotations(Symbol symbol) {
+  public static Stream<? extends AnnotationMirror> getTypeUseAnnotations(Symbol symbol) {
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java:274]: Why the change to public?

**Review by msridhar — APPROVED** [review]
Approved with a minor question

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java (hunk 2 above)
Comment (lazaroclapp) [reply nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java:274]: Oh, for debugging. I just missed reverting this one. Good catch! :)
